### ISSUE TITLE
SpacyRecognizer supported_entities not respected in Analyze method

### ISSUE THREAD

**Issue by jperuggia** [microsoft/presidio#1216]
**Describe the bug**
When using the SpacyRecognizer class and setting the property of `supported_entities`, the recognizer returns results that are not valid for the list of supported entities. This breaking change was introduced from version >= 2.3.34.  

**To Reproduce**

1. Create and modify the registry to remove the default SpacyRecognizer via
```
registry = RecognizerRegistry()
registry.load_predefined_recognizers()
registry.remove_recognizer("SpacyRecognizer")
sr = SpacyRecognizer(supported_entities=["DATE_TIME", "NRP"])
registry.add_recognizer(sr)
```
2. Create the analyzer using this new registry
```
analyzer = AnalyzerEngine(
    registry=registry,
    supported_languages='en'
)
```        

3. Run text with a Person or entity that isn't in the list of supported_entities as indicated above. That entity will still be detected and identified via the SpacyRecognizer. 

**Expected behavior**
The SpacyRecognizer should not return results that are not in the list of supported_entities

**Additional context**
In version `2.2.33` the library had a [line](https://github.com/microsoft/presidio/blob/b6f81912e9b3649c39b341c6d8f203afaf60087e/presidio-analyzer/presidio_analyzer/predefined_recognizers/spacy_recognizer.py#L98C13-L99C25) which filtered out these entities. 

Starting in version `2.2.34` this filter is no longer present and appears to be erroneously removed.

**Comment by omri374** [comment]
Thanks for your feedback! We will look into this.

**Comment by omri374** [comment]
Hi @jperuggia, I couldn't reproduce this. Are you using the latest version of `SpacyRecognizer`? 
There's still a line in `SpacyRecognizer` which checks for the list of entities:
https://github.com/microsoft/presidio/blob/8b9d961d1f4ddb65e124a480f3c72c63894b0b5f/presidio-analyzer/presidio_analyzer/predefined_recognizers/spacy_recognizer.py#L106

The list of entities is passed from the AnalyzerEngine, which in turn takes it from the recognizer registry. If the `SpacyRecognizer` is configured to support only `NRP` and `DATE_TIME`, then the recognizer registry would not contain `PERSON` as a supported entity, and this entity would not be passed to `SpacyRecognizer.analyze`

Could you please provide more details on how to reproduce this?

**Comment by jperuggia** [comment]
Here is a full example showing the error

```
    from presidio_analyzer.nlp_engine import NlpEngineProvider
    from presidio_analyzer.predefined_recognizers import SpacyRecognizer
    from presidio_analyzer import AnalyzerEngine, RecognizerRegistry, RecognizerResult

    languages_config = {
        "nlp_engine_name": "spacy",
        "models": [
            {"lang_code": "en", "model_name": "en_core_web_trf"},
        ],
    }

    provider = NlpEngineProvider(nlp_configuration=languages_config)
    nlp_engine = provider.create_engine()

    registry = RecognizerRegistry()
    registry.load_predefined_recognizers()
    registry.remove_recognizer("SpacyRecognizer")
    sr = SpacyRecognizer(supported_entities=["NRP"])
    registry.add_recognizer(sr)

    engine = AnalyzerEngine(
        registry=registry,
        supported_languages=['en'],
        nlp_engine=nlp_engine,
    )

    # Here is a sample text with some entities.
    sample_text = "My name is John Smith and I live in NYC. I moved here in the early 2000's. I like to get pizza from Joe's Pizza."

    # call analyzer engine and see the results.
    anon_results = engine.analyze(
        text=sample_text,
        entities=['PERSON', 'NRP', 'DATE_TIME', 'LOCATION', 'CREDIT_CARD'], # this is just a sample list
        language='en',
        allow_list=None
    )


    print('anon result inspect')
   ```



If we look at the values in the `anon_results` array, we see that there is something that is reconigzed as a Person which is identified via the SpacyRecognizer. 

![image](https://github.com/microsoft/presidio/assets/1916227/3f1ffb6d-060f-47ca-88fb-df5eea0c5c9c)


If we look at the configured item in the registry only NRP entities should be supported by Spacy at this time.

![image](https://github.com/microsoft/presidio/assets/1916227/ccbb07d0-f647-4a49-8d04-03edc784b114)

**Comment by omri374** [comment]
I see now. The call to `AnalyzerEngine.analyze` contains an `entities` parameter, which contains PERSON. Without it, it wouldn't return a PERSON as the recognizer is not configured to return it. It's indeed a bug, but a simple workaround would be not to put `PERSON` in the list of entities for the `AnalyzerEngine.analyze` method, or not pass any entity names at all.

**Comment by omri374** [comment]
Fixed in #1221
